src/action.ts

```diff
@@ -123,40 +124,86 @@ export async function performAction(
 	tracker: string
 ): Promise<ActionResult> {
 	const { action, linkDir } = getRuntimeConfig();
-	let linkedFilesRoot: string | undefined;
 
-	if (linkDir) {
-		const linkedFilesRootResult = await linkAllFilesInMetafile(
+	// SAVES FIRST, PREVENTS LINKING THOUGH IN THIS ORDER
+	if (action === Action.SAVE) {
+		saveTorrentFile(tracker, getTag(searchee.name), newMeta);
+		const styledName = chalk.green.bold(newMeta.name);
+		const styledTracker = chalk.bold(tracker);
+		logger.info(`Found ${styledName} on ${styledTracker} - saved`);
+		return SaveResult.SAVED;
+	}
+
+	let destinationDir: string | undefined,
+		ogDownloadDir,
+		linkedFilesRootResult;
+
+	// SET THE CLIENTS SAVE PATH - PREVENTS ERROR IF DATADIR/LINKDIR IS SET
+	// BUT NOT THE TORRENT DATA DIRECTORY AND/OR LINKING THROWS OR IS NOT SET
+	if (searchee.infoHash) {
+		ogDownloadDir = await getClient().getDownloadDir(searchee);
```

Comment:
do we need ogDownloadDir to be defined up on line 138? I see some commented code below that references it, not sure what the status of that is. If it's unnecessary, I think `ogDownloadDir` could be changed to a const inside this `if` block.
```suggestion
		const ogDownloadDir = await getClient().getDownloadDir(searchee);
```

Reply:
That commented code was for reference from the old previous code, it's irrelevant now.

It can be.